backend/src/LibExecution/ProgramTypesToRuntimeTypes.fs

```diff
@@ -179,57 +179,50 @@ module Expr =
-      // from the OCaml interpreter
-      let inner = toRT expr1
-
-      List.fold
-        inner
-        (fun prev next ->
-          let rec convert thisExpr =
-            match thisExpr with
-            // TODO: support currying
-
-            | PT.EFnCall (id, fnName, typeArgs, PT.EPipeTarget ptID :: exprs) ->
-              RT.EApply(
-                id,
-                RT.FnName(FQFnName.toRT fnName),
-                List.map TypeReference.toRT typeArgs,
-                prev :: List.map toRT exprs
-              )
-            | PT.EInfix (id, PT.InfixFnCall fnName, PT.EPipeTarget ptID, expr2) ->
-              let (modules, fn, version) = InfixFnName.toFnName fnName
-              let name =
-                PT.FQFnName.Stdlib(
-                  { modules = modules; function_ = fn; version = version }
-                )
-              let typeArgs = []
-              RT.EApply(
-                id,
-                RT.FnName(FQFnName.toRT name),
-                typeArgs,
-                [ prev; toRT expr2 ]
-              )
-            // Binops work pretty naturally here
-            | PT.EInfix (id, PT.BinOp op, PT.EPipeTarget _, expr2) ->
-              match op with
-              | PT.BinOpAnd -> RT.EAnd(id, prev, toRT expr2)
-              | PT.BinOpOr -> RT.EOr(id, prev, toRT expr2)
-            | other ->
-              let typeArgs = [] // TODO: review
-              RT.EApply(pipeID, RT.FnTargetExpr(toRT other), typeArgs, [ prev ])
-          convert next)
-
-        (expr2 :: rest)
+      let folder (prev : RT.Expr) (next : PT.PipeExpr) : RT.Expr =
+
+        let applyFn (expr : RT.Expr) =
+          let typeArgs = []
+          RT.EApply(pipeID, RT.FnTargetExpr(expr), typeArgs, [ prev ])
+
+        match next with
+        | PT.EPipeFnCall (id, fnName, typeArgs, exprs) ->
+          RT.EApply(
+            id,
+            RT.FnName(FQFnName.toRT fnName),
+            List.map TypeReference.toRT typeArgs,
+            prev :: List.map toRT exprs
+          )
+        | PT.EPipeInfix (id, PT.InfixFnCall fnName, expr) ->
+          let (modules, fn, version) = InfixFnName.toFnName fnName
+          let name =
+            PT.FQFnName.Stdlib(
+              { modules = modules; function_ = fn; version = version }
+            )
+          let typeArgs = []
+          RT.EApply(id, RT.FnName(FQFnName.toRT name), typeArgs, [ prev; toRT expr ])
+        // Binops work pretty naturally here
+        | PT.EPipeInfix (id, PT.BinOp op, expr) ->
+          match op with
+          | PT.BinOpAnd -> RT.EAnd(id, prev, toRT expr)
+          | PT.BinOpOr -> RT.EOr(id, prev, toRT expr)
+        | PT.EPipeEnum (id, typeName, caseName, _) ->
+          RT.EEnum(id, FQTypeName.toRT typeName, caseName, [ prev ])
+        | PT.EPipeVariable (id, name) -> RT.EVariable(id, name) |> applyFn
+        | PT.EPipeFieldAccess (id, obj, fieldname) ->
+          RT.EFieldAccess(id, toRT obj, fieldname) |> applyFn
```

Comment:
I don't think there's any tests for EFieldAccess

Reply:
I think it would be better to remove EPipeField access for now. We don't have parsing or tests and it's a new feature.